raft/raft.cpp

```diff
@@ -792,49 +884,229 @@ raft::load_state()
 
 
 void
-raft::load_log_entries()
+raft::perform_commit(uint32_t& commit_index, const bzn::log_entry& log_entry)
 {
-    std::ifstream is(this->entries_log_path(), std::ios::in | std::ios::binary);
-    bzn::log_entry log_entry;
-    while (is >> log_entry)
+    this->notify_commit(commit_index, log_entry.json_to_string(log_entry.msg));
+    this->commit_handler(log_entry.msg);
+    commit_index++;
+    this->save_state();
+
+    if(this->get_state() == bzn::raft_state::leader && log_entry.entry_type == bzn::log_entry_type::joint_quorum)
     {
-        this->log_entries.emplace_back(log_entry);
+        this->append_log_unsafe(
```

Comment:
you lock elsewhere. why not here?

Reply:
This is where we would get thread locking if we used the "safe" version as we are already within the scope of 
     std::lock_guard<std::mutex> lock(this->raft_lock)
from the top of handle_ws_raft_messages.